### PR TITLE
[stdlib] Add Swift 3 compatibility for String.init from old integer protocols

### PR DESCRIPTION
**Pull request by airspeedswift**
Adds Swift-4-obsoleted versions that to signed/unsigned without them being fixed-width.

rdar://problem/31727219

### CODE REVIEW

**Comment by airspeedswift** [comment]
@swift-ci Please test

File: stdlib/public/core/StringLegacy.swift [hunk 1]
```diff
@@ -285,6 +285,45 @@ extension String {
...
+  ///   - radix: The base to use for the string representation. `radix` must be
+  ///     at least 2 and at most 36. The default is 10.
+  ///   - uppercase: Pass `true` to use uppercase letters to represent numerals
+  ///     greater than 9, or `false` to use lowercase letters. The default is
+  ///     `false`.
+  // FIXME(integers): tiebreaker between T : FixedWidthInteger and other obsoleted
+  @available(swift, obsoleted: 4)
+  public init<T : FixedWidthInteger>(
+    _ value: T, radix: Int = 10, uppercase: Bool = false
+  ) where T : SignedInteger {
```
Comment (moiseev) [inline stdlib/public/core/StringLegacy.swift]: What if you add this line to the non-obsolete version on `FixedWIdthInteger`?

Comment (airspeedswift) [inline stdlib/public/core/StringLegacy.swift]: That would be detrimental to functionality, no? Someone who wants to write an extension on FixedWidthInteger that used this function would need to also constrain their function to either Signed or UnsignedInteger

**Comment by airspeedswift** [comment]
@swift-ci Please smoke test

**Pull request merged**